### PR TITLE
Adds new render option

### PR DESCRIPTION
**Pull request by gtomitsuka**
I developed this function for myself because I need it for a project I'm working on, but it basically adds a way for `finalhandler`'s users' to do custom rendering of their views.

The `test/test.js` passes, and I think the README.md and test/test.js show how to work with it properly.

### CODE REVIEW

**Comment by dougwilson** [comment]
This module does not do much besides simply render. Why even invoke this module at all instead of defining error handlers in Express?

File: index.js [hunk 1]
```diff
@@ -104,7 +106,12 @@ function finalhandler(req, res, options) {
       return req.socket.destroy()
     }
 
-    send(req, res, status, msg)
+    if(typeof render === 'function'){
```
Comment (dougwilson) [inline index.js]: Calling `finalhandler(req, res, {render: 42})` needs to throw an error, not silently pretend like the option was not there.

File: index.js [hunk 2]
```diff
@@ -104,7 +106,12 @@ function finalhandler(req, res, options) {
...
+      render(msg, req, res, function(rendered){
```
Comment (dougwilson) [inline index.js]: The signature `function(rendered)` is not acceptable here; it means an error could never possibly occurs within rendering and does not property fit in with error-first callback pattern. Please have it accept `err` as a first argument and then act on that error if one is provided.

File: README.md [hunk 3]
```diff
@@ -7,6 +7,7 @@
 [![Test Coverage][coveralls-image]][coveralls-url]
 
 Node.js function to invoke as the final step to respond to HTTP request.
+***Note:*** This is a fork maintained by [@gtomitsuka](https://github.com/gtomitsuka), and includes a `render` option, for custom rendering.
```
Comment (dougwilson) [inline README.md]: Please remove this note.

**Comment by dougwilson** [comment]
So besides the question and the two notes, I am working on getting [the 1.x branch ported to master](https://github.com/pillarjs/finalhandler/tree/1.x) for the Express 5.0 release next week. I updated the 1.x branch to be based on 0.4.0 and already moved one of the commits onto master. If you can possibly rebase this PR onto the old 1.x branch (I see you have that branch in your fork already), that would help a lot as well :)!

**Comment by gtomitsuka** [comment]
Oops :P

**Comment by gtomitsuka** [comment]
@dougwilson I would like usable 404 handlers working without adding additional handlers, and I think other people want it, too. And I'll make the changes.

**Comment by dougwilson** [comment]
> I would like usable 404 handlers working without adding additional handlers

What does that mean? Remember, this module is what Express uses _only if you don't provide anything_. Are you sure you are not imposing Express questions/issues onto this module?

**Comment by dougwilson** [comment]
The other weird part of your statement regarding 404 handlers here is that the `render` option you have in this PR only works for errors; it doesn't have anything to do with 404 handling. Perhaps your render is working on the wrong part? I'm really, really confused and I'm probably going to say that I need you to provide your full, complete ultimate use-case you are trying to achieve with this PR before we can proceed here so I know this is the right solution.

**Comment by gtomitsuka** [comment]
First of all, for this one project I am using finalhandler() directly.

Now, the render works for both errors and 404s. When calling `done()` without an error, the message appears:

screen shot 2015-07-17 at 12 04 48 pm

I use finalhandler for both errors and 404 messages. With the render option 

I thought of it because I wanted a catcher for 404 errors, but it's not _only_ for 404 errors. With the render function people will be able to actually use the Express/Connect error messages(like Express' try catch) on production.

**Comment by dougwilson** [comment]
Gotcha. I don't see a test in your PR to test using render with 404s. Can you add it?

**Comment by gtomitsuka** [comment]
Sure.

**Comment by gtomitsuka** [comment]
(I'm busy today, I'll do it tomorrow)

**Comment by dougwilson** [comment]
So there hasn't been any update for a couple years and the original repo is gone, so I cannot merge. Please feel free to open it up again if you would like to resume.

**Pull request closed without merging**